api/gloo/gateway/v1/gateway.proto

```diff
@@ -0,0 +1,134 @@
+syntax = "proto3";
+package gateway.solo.io;
+option go_package = "github.com/solo-io/solo-apis/pkg/api/gateway.solo.io/v1";
+
+import "google/protobuf/struct.proto";
+import "gogoproto/gogo.proto";
+option (gogoproto.equal_all) = true;
+import "extproto/ext.proto";
+option (extproto.hash_all) = true;
+
+
+import "google/protobuf/wrappers.proto";
+import "solo-kit/api/v1/ref.proto";
+import "solo-kit/api/v1/solo-kit.proto";
+
+import "solo-apis/api/gloo/gloo/v1/proxy.proto";
+import "solo-apis/api/gloo/gloo/v1/options.proto";
+
+/*
+A Gateway describes a single Listener (bind address:port)
+and the routing configuration to upstreams that are reachable via a specific port on the Gateway Proxy itself.
+*/
+message GatewaySpec {
+
+
+    // if set to false, only use virtual services without ssl configured.
+    // if set to true, only use virtual services with ssl configured.
+    bool ssl = 1;
+
+    // the bind address the gateway should serve traffic on
+    string bind_address = 3;
+
+    // bind ports must not conflict across gateways for a single proxy
+    uint32 bind_port = 4;
+
+    // top level optional configuration for all routes on the gateway
+    gloo.solo.io.ListenerOptions options = 5;
+
+    // Status indicates the validation status of this resource.
+    // Status is read-only by clients, and set by gloo during validation
+
+    // Metadata contains the object metadata for this resource
```

Comment:
we can almost-but-not-quite  count on these comments being exactly the same on each resource. It doesn't really matter but i'm worried about these comments applying to the next-occurring field, use_proxy_proto in this case.

I wonder if it would be worth rounding up all possible comment strings and doing a `sed` to remove them from the outputted protos. wdyt?

Reply:
If they are the same I could just not add them

Reply:
They don't show up on the use_proxy_proto .pb.go file, so I think we're safe here.